src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/domainEvents/ComAllocatedHandler.kt

```diff
@@ -9,68 +9,134 @@ import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.model.request.Updat
+        firstName = offenderManager.forename,
+        lastName = offenderManager.surname,
       ),
     )
 
-    offenderService.updateOffenderWithResponsibleCom(crn, newCom)
+    offenderService.updateOffenderWithResponsibleCom(offenderManager.crn, newCom)
 
     offenderService.updateProbationTeam(
-      crn,
+      offenderManager.crn,
       UpdateProbationTeamRequest(
-        probationAreaCode = offenderManager.provider.code,
-        probationAreaDescription = offenderManager.provider.description,
-        probationPduCode = offenderManager.team.borough.code,
-        probationPduDescription = offenderManager.team.borough.description,
-        probationLauCode = offenderManager.team.district.code,
-        probationLauDescription = offenderManager.team.district.description,
-        probationTeamCode = offenderManager.team.code,
-        probationTeamDescription = offenderManager.team.description,
+        probationAreaCode = offenderManager.providerCode,
+        probationAreaDescription = offenderManager.providerDescription,
+        probationPduCode = offenderManager.boroughCode,
+        probationPduDescription = offenderManager.boroughDescription,
+        probationLauCode = offenderManager.districtCode,
+        probationLauDescription = offenderManager.districtDescription,
+        probationTeamCode = offenderManager.teamCode,
+        probationTeamDescription = offenderManager.teamDescription,
       ),
     )
   }
+
+  fun isValidOffenderManager(offenderManager: OffenderManager?, crn: String): Boolean {
+    if (offenderManager == null) {
+      log.warn("No offender manager found for CRN $crn")
+      return false
+    }
+    if (offenderManager.username.isNullOrBlank()) {
+      log.warn("Offender manager with code ${offenderManager.code} for CRN $crn has no username — skipping allocation")
+      return false
+    }
+    return true
+  }
+
+  private fun getOffenderManager(crn: String, event: HMPPSDomainEvent): OffenderManager? {
+    val isDeliusEvent = event.detailUrl.isNullOrBlank()
+    return if (isDeliusEvent) {
+      getOffenderManagerForDeliusEvent(crn)
+    } else {
+      getOffenderManagerForApop(event.detailUrl!!)
+    }
+  }
+
+  fun getOffenderManagerForDeliusEvent(crn: String): OffenderManager? {
+    log.info("Getting offender manager from Delius for CRN: {}", crn)
+
+    val response = deliusApiClient.getOffenderManager(crn)
+    return if (response != null) {
+      offenderManagerMapper.mapFrom(response)
+    } else {
+      log.warn("No Delius data found for CRN: {}", crn)
+      null
+    }
+  }
+
+  private fun getOffenderManagerForApop(detailUrl: String): OffenderManager? {
+    val personUuid = getPersonUuid(detailUrl)
+    log.info("Getting offender manager from CVL for personUuid: {}", personUuid)
+
+    val workLoadAllocationResponse = workLoadApiClient.getStaffDetails(personUuid)
+    if (workLoadAllocationResponse == null) {
+      log.warn("No staff codes for personUuid: {}", personUuid)
```

Comment:
not sure this is a realistic scenario.. think just letting it throw would make sense here as we should be alerted to this and investigate fix/any issue.

Reply:
if it throws would it put the message back on the Queue or DLQ